src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -37,8 +38,6 @@ final class KeyMatcher {
     private static final int SKIP_KEY_LOOKUP = -1;
     private final JsonMaskingConfig maskingConfig;
     private final TrieNode root;
-    // used for an optimization to remember all key length and return early if the key length is not known
-    private final boolean[] knownKeyLengthsInBytes = new boolean[256];
```

Comment:
Can't use this (micro-)optimization anymore as every input character of 1-4 bytes can be matching 1-12 bytes of json key if that happens to be encoded.